File: JAVA/level1/소수_찾기.java
class Solution {
  public int solution(int n) {
      int answer = 0;
      int[] array = new int[n+1];

      for(int i=0; i<array.length; i++)
          array[i] = i;

      for(int i=2; i<array.length; i++)
          if(array[i] != 0)
              for(int j=i+array[i]; j<array.length; j+=array[i])
                  array[j] = 0;

      for(int i=2; i<array.length; i++)
          if(array[i] != 0)
              answer++;

      return answer;
  }
}

File: JavaScript/level1/소수찾기.js
function solution(n) {
    var answer = 0;
    let array = Array(n+1).fill(false); 
    
    for(let i=2; i<=n; i++)
        if(array[i] === false)
            for(let j=i*2; j<=n; j=j+i)
                array[j] = true;
    
    for(let i=2; i<=n; i++)
        if(array[i] === false)
            answer++;

    return answer;
}
